tests/server/test_desired_state_versions.py

```diff
@@ -0,0 +1,323 @@
+"""
+    Copyright 2021 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import datetime
+import json
+import uuid
+from operator import itemgetter
+from typing import Dict, List, Tuple
+
+import pytest
+from tornado.httpclient import AsyncHTTPClient, HTTPRequest
+
+from inmanta import data
+from inmanta.data.model import DesiredStateLabel
+from inmanta.server.config import get_bind_port
+
```

Comment:
Could you also add a test to verify that a `404` is returned when `list_desired_state_versions` endpoint is called using a non-existing environment?